assets/react/v3/entries/course-builder/components/additional/MeetingForm.tsx

```diff
@@ -0,0 +1,273 @@
+import { css } from '@emotion/react';
+import { __ } from '@wordpress/i18n';
+import { Controller } from 'react-hook-form';
+
+import Button from '@Atoms/Button';
+
+import FormCheckbox from '@Components/fields/FormCheckbox';
+import FormDateInput from '@Components/fields/FormDateInput';
+import FormInput from '@Components/fields/FormInput';
+import FormTextareaInput from '@Components/fields/FormTextareaInput';
+import FormTimeInput from '@Components/fields/FormTimeInput';
+
+import { borderRadius, colorPalate, colorTokens, fontSize, shadow, spacing } from '@Config/styles';
+import { typography } from '@Config/typography';
+import Show from '@Controls/Show';
+
+import { useFormWithGlobalError } from '@Hooks/useFormWithGlobalError';
+
+import { Meeting } from './LiveClass';
+import { styleUtils } from '@Utils/style-utils';
+import FormSelectInput from '@Components/fields/FormSelectInput';
+
+export type MeetingType = 'zoom' | 'google_meet' | 'jitsi';
+
+interface MeetingFormFieldProps {
+  meeting_name: string;
+  meeting_summary: string;
+  meeting_date: string;
+  meeting_time_from: string;
+  meeting_time_to: string;
+  meeting_enrolledAsAttendee: boolean;
+  meeting_autoRecording: string;
+  meeting_password: string;
+  meeting_host: string;
+}
+
+interface MeetingFormProps {
+  type: MeetingType;
+  setShowMeetingForm: React.Dispatch<React.SetStateAction<MeetingType | null>>;
+  setMeetings: React.Dispatch<React.SetStateAction<Meeting[]>>;
+}
+
+const MeetingForm = ({ type, setShowMeetingForm, setMeetings }: MeetingFormProps) => {
+  const meetingForm = useFormWithGlobalError<MeetingFormFieldProps>({
+    defaultValues: {
+      meeting_name: '',
+      meeting_summary: '',
+      meeting_date: '',
+      meeting_time_from: '',
+      meeting_time_to: '',
+      meeting_enrolledAsAttendee: false,
+      meeting_autoRecording: 'disabled',
+      meeting_password: '',
+      meeting_host: '',
+    },
+  });
+
+  const onCancel = () => {
+    setShowMeetingForm(null);
+  };
+
+  // @TODO: will come from app config api later.
+  const onSubmit = (data: MeetingFormFieldProps) => {
+    setShowMeetingForm(null);
+    meetingForm.reset();
+    let dataToSubmit: Omit<Meeting, 'id'>;
+
+    if (!data.meeting_date || !data.meeting_time_from || !data.meeting_time_to) {
+      return;
+    }
+
+    if (type === 'google_meet') {
+      dataToSubmit = {
+        type: type,
+        meeting_title: data.meeting_name,
+        meeting_date: data.meeting_date,
+        meeting_start_time: data.meeting_time_from,
+        meeting_link: 'https://meet.google.com/abc-xyz',
+      };
+    } else if (type === 'zoom') {
+      dataToSubmit = {
+        type: type,
+        meeting_title: data.meeting_name,
+        meeting_date: data.meeting_date,
+        meeting_start_time: data.meeting_time_from,
+        meeting_link: 'https://zoom.us/abc-xyz',
+        meeting_token: 'abc-xyz',
+        meeting_password: data.meeting_password,
+      };
+    }
```

Comment:
Most of the fields are the same here. We could achieve this simple declaring one object.
```suggestion
const dataToSubmit = {
  type: type,
  meeting_title: data.meeting_name,
  meeting_date: data.meeting_date,
  meeting_start_time: data.meeting_time_from,
  meeting_link: type === 'zoom' ? 'https://zoom.us/abc-xyz': 'https://meet.google.com/abc-xyz',
  ...(type === 'zoom' && {
    meeting_token: 'abc-xyz'
    meeting_password: data.meeting_password,
  }),
};
```